File: happykopiAPI/happykopiAPI/DTOs/Product/Incoming Data/ProductVariantCreateDto.cs
using System.ComponentModel.DataAnnotations;

namespace happykopiAPI.DTOs.Product.Incoming_Data
{
    public class ProductVariantCreateDto
    {
        [Required]
        public string Size { get; set; }

        [Required]
        [Range(0.01, double.MaxValue, ErrorMessage = "Price must be greater than 0")]
        public decimal Price { get; set; }

        public ICollection<ProductVariantIngredientCreateDto> Recipe { get; set; } = [];
        public ICollection<ProductVariantAddOnCreateDto> AddOns { get; set; } = [];
    }
}

File: happykopiApp/src/app/core/dtos/product/product-variant-create-dto.ts
import { ProductVariantAddOnCreateDto } from "./product-variant-add-on-create-dto";
import { ProductVariantIngredientCreateDto } from "./product-variant-ingredient-create-dto";

export interface ProductVariantCreateDto {
    SizeId: number;
    Price: number;
    Recipe: ProductVariantIngredientCreateDto[];
    AddOns: ProductVariantAddOnCreateDto[];
}
